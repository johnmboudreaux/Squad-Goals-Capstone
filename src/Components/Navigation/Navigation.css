.navigation {
  background: rgba(87, 162, 205, 0.05);
  height: 100px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center; }

.navigation .nav-button {
  height: 40px; }

.navigation h1 {
  align-items: center;
  color: #585858;
  display: flex;
  font-size: 2.5em;
  justify-content: center;
  margin: 0;
  margin-left: 40px;
  width: 35%; }

.navigation ul {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  width: 65%;
  height: 40px; }

.nav-links {
  list-style: none; }

.nav-links a {
  display: block;
  font-size: 1.2em;
  margin-right: 40px; }
