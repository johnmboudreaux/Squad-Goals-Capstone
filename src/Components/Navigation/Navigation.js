import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { auth } from '../../Utils/fire';

class Navigation extends Component {
  constructor() {
    super();
    this.state = {
      showMyComponent: false,
    };
  }

  componentWillMount() {
    this.checkForUser();
  }

  checkForUser() {
    auth.onAuthStateChanged((user) => {
      user ?
        this.setState({ showMyComponent: true })
        :
        this.setState({ showMyComponent: false });
    });
  }

  logout() {
    auth.signOut()
      .then(res => res)
      .catch(error => console.error('log out error', error));
  }

  render() {
    return (
      <nav className="navigation">
        <ul className="">
          <li className="nav-links">
            <Link
              style={this.state.showMyComponent ? { display: 'block' } : { display: 'none' }}
              onClick={() => {
              this.logout();
              this.checkForUser();
            }}
              to="/"
            >Log Out
            </Link>
          </li>
          <li className="nav-links">
            <Link to="/UserDashboard">Dashboard</Link>
          </li>
        </ul>
      </nav>
    );
  }
}

export default Navigation;
