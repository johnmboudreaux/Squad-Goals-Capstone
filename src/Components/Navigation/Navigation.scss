.navigation {
  background-color: $main;
  height: 7vh;
}

.navigation ul {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.nav-links {
  list-style: none;
}

.nav-links a {
  display: block;
  font-size: 1.2em;
  margin-top: 15px;
}
