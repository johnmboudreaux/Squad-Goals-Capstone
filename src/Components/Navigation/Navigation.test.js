import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Navigation from './Navigation';
// import goal from '../../../data/test/goals_data_test';

describe('Navigation snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<Navigation />);

    expect(wrapper).toMatchSnapshot();
  });


  // it('renders without crashing', () => {
  //   const div = document.createElement('div');
  //   ReactDOM.render(<Navigation />, div);
  // });
});
