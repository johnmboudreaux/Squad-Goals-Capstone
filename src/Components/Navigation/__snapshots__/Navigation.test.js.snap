// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className=""
      >
        <li
          className="nav-links"
        >
          <Link
            onClick={[Function]}
            replace={false}
            style={
              Object {
                "display": "none",
              }
            }
            to="/"
          >
            Log Out
          </Link>
        </li>
        <li
          className="nav-links"
        >
          <Link
            replace={false}
            to="/UserDashboard"
          >
            Dashboard
          </Link>
        </li>
      </ul>,
      "className": "navigation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-links"
          >
            <Link
              onClick={[Function]}
              replace={false}
              style={
                Object {
                  "display": "none",
                }
              }
              to="/"
            >
              Log Out
            </Link>
          </li>,
          <li
            className="nav-links"
          >
            <Link
              replace={false}
              to="/UserDashboard"
            >
              Dashboard
            </Link>
          </li>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              onClick={[Function]}
              replace={false}
              style={
                Object {
                  "display": "none",
                }
              }
              to="/"
            >
              Log Out
            </Link>,
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Log Out",
              "onClick": [Function],
              "replace": false,
              "style": Object {
                "display": "none",
              },
              "to": "/",
            },
            "ref": null,
            "rendered": "Log Out",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/UserDashboard"
            >
              Dashboard
            </Link>,
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Dashboard",
              "replace": false,
              "to": "/UserDashboard",
            },
            "ref": null,
            "rendered": "Dashboard",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className=""
        >
          <li
            className="nav-links"
          >
            <Link
              onClick={[Function]}
              replace={false}
              style={
                Object {
                  "display": "none",
                }
              }
              to="/"
            >
              Log Out
            </Link>
          </li>
          <li
            className="nav-links"
          >
            <Link
              replace={false}
              to="/UserDashboard"
            >
              Dashboard
            </Link>
          </li>
        </ul>,
        "className": "navigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="nav-links"
            >
              <Link
                onClick={[Function]}
                replace={false}
                style={
                  Object {
                    "display": "none",
                  }
                }
                to="/"
              >
                Log Out
              </Link>
            </li>,
            <li
              className="nav-links"
            >
              <Link
                replace={false}
                to="/UserDashboard"
              >
                Dashboard
              </Link>
            </li>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                onClick={[Function]}
                replace={false}
                style={
                  Object {
                    "display": "none",
                  }
                }
                to="/"
              >
                Log Out
              </Link>,
              "className": "nav-links",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log Out",
                "onClick": [Function],
                "replace": false,
                "style": Object {
                  "display": "none",
                },
                "to": "/",
              },
              "ref": null,
              "rendered": "Log Out",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/UserDashboard"
              >
                Dashboard
              </Link>,
              "className": "nav-links",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Dashboard",
                "replace": false,
                "to": "/UserDashboard",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
